generic.go

```diff
@@ -95,21 +74,75 @@ func NewGeneric(config *Config, queue workqueue.RateLimitingInterface, workers i
 				Help:      fmt.Sprintf("Histogram measuring the time it took to process a %s", &groupKind),
 			},
 		)
+		// Extra api data
+		requestCount := prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: fmt.Sprintf("processed_%s_count", objectName),
+				Help: fmt.Sprintf("Cumulative number of %s processed", &groupKind),
+			},
+			[]string{"url", "method", "status"},
+		)
+		requestTime := prometheus.NewHistogram(
+			prometheus.HistogramOpts{
+				Name: fmt.Sprintf("processed_%s_time", objectName),
+				Help: fmt.Sprintf("Number of seconds each request to %s takes", &groupKind),
+			},
+		)
+
 		allMetrics := []prometheus.Collector{
 			objectProcessCount, objectProcessTime,
+			requestCount, requestTime,
 		}
 		for _, metric := range allMetrics {
 			if err := constructorConfig.Registry.Register(metric); err != nil {
 				return nil, errors.WithStack(err)
 			}
 		}
 
-		holders[descr.Gvk] = Holder{
-			Cntrlr:             iface,
-			ZapNameField:       descr.ZapNameField,
-			ReadyForWork:       readyForWork,
-			objectProcessCount: objectProcessCount,
-			objectProcessTime:  objectProcessTime,
+		constructed, err := constr.New(
+			constructorConfig,
+			&Context{
+				ReadyForWork: func() {
+					close(readyForWork)
+				},
+				Middleware:  addMiddleware(requestCount, requestTime),
```

Comment:
Random thought : using the ctrl library requires a lot of knowledge about how ctrl works. I mean it's a bit easy to forget to call ReadyForWork and to attach the middleware to your handler.